### PR TITLE
IGNITE-13071: Improve test coverage of cluster read-only mode.

### PR DESCRIPTION
**Pull request by antonovsergey93**
Thank you for submitting the pull request to the Apache Ignite.

In order to streamline the review of the contribution 
we ask you to ensure the following steps have been taken:

### The Contribution Checklist
- [x] There is a single JIRA ticket related to the pull request. 
- [x] The web-link to the pull request is attached to the JIRA ticket.
- [x] The JIRA ticket has the _Patch Available_ state.
- [x] The pull request body describes changes that have been made. 
The description explains _WHAT_ and _WHY_ was made instead of _HOW_.
- [x] The pull request title is treated as the final commit message. 
The following pattern must be used: `IGNITE-12407: Add Cluster API support to Java thin client`
- [x] A reviewer has been mentioned through the JIRA comments 
(see [the Maintainers list](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute#HowtoContribute-ReviewProcessandMaintainers)) 
- [x] The pull request has been checked by the Teamcity Bot and 
the `green visa` attached to the JIRA ticket (see [TC.Bot: Check PR](https://mtcga.gridgain.com/prs.html))

### Notes
- [How to Contribute](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute)
- [Coding abbreviation rules](https://cwiki.apache.org/confluence/display/IGNITE/Abbreviation+Rules)
- [Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines)
- [Apache Ignite Teamcity Bot](https://cwiki.apache.org/confluence/display/IGNITE/Apache+Ignite+Teamcity+Bot)

If you need any help, please email dev@ignite.apache.org or ask anу advice on http://asf.slack.com _#ignite_ channel.

### CODE REVIEW

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/ClusterCachesInfo.java [hunk 1]
```diff
@@ -960,6 +962,21 @@ private boolean processStartNewCacheRequest(
         DynamicCacheChangeRequest req,
         String cacheName
     ) {
+        CacheConfiguration<?, ?> ccfg = req.startCacheConfiguration();
+
+        if (ctx.cache().context().readOnlyMode()) {
```
Comment (zstan) [inline modules/core/src/main/java/org/apache/ignite/internal/processors/cache/ClusterCachesInfo.java:967]: code copy-paste is always evil )

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/GridCacheAdapter.java [hunk 2]
```diff
@@ -4759,6 +4763,8 @@ private boolean clearLocally0(K key, boolean readers) {
 
         ctx.checkSecurity(SecurityPermission.CACHE_REMOVE);
 
+        ctx.shared().cache().checkReadOnlyState("clear", ctx.config());
```
Comment (zstan) [inline modules/core/src/main/java/org/apache/ignite/internal/processors/cache/GridCacheAdapter.java:4766]: I suppose no neet to check mvcc and security permissions in already readonly cluster ? may be we need to move it upper ?

File: modules/core/src/test/java/org/apache/ignite/internal/processors/cache/CacheCreateDestroyClusterReadOnlyModeTest.java [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+            for (Ignite node : G.allGrids()) {
+                executeAndCheckException(() -> node.createCache(cfg), cfg.getName());
+                executeAndCheckException(() -> node.createCache(cfg.getName()), cfg.getName());
+            }
+        }
+    }
+
+    /** */
+    @Test
+    public void testGetOrCreateNotExistedCacheDenied() {
```
Comment (zstan) [inline modules/core/src/test/java/org/apache/ignite/internal/processors/cache/CacheCreateDestroyClusterReadOnlyModeTest.java:58]: copy paste of testCacheCreateDenied or i miss something ?

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/ClusterCachesInfo.java (hunk 1 above)
Comment (antonovsergey93) [reply modules/core/src/main/java/org/apache/ignite/internal/processors/cache/ClusterCachesInfo.java:967]: Fixed

**Pull request closed without merging**